keyboards/hfdkb/monsgeek/m1/matrix.c

```diff
@@ -0,0 +1,286 @@
+/* Copyright (C) 2022 jonylee@hfd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "eeconfig.h"
+#include "quantum.h"
+
+enum __layers {
+    WIN_B,
+    WIN_WASD,
+    WIN_FN,
+    MAC_B,
+    MAC_WASD,
+    MAC_FN
+};
+
+enum colors { WHITE, RED, GREEN, BLUE };
+enum colors led_color_status = WHITE;
+
+static bool     fn_make_flag        = false;
+static bool     Lkey_flag           = false;
+static bool     reset_glint_flag    = false;
+static bool     while_test_flag     = false;
+static bool     alarm_flag          = false;
+static uint16_t current_time        = 0;
+static uint8_t  glint_cnt           = 0;
+static uint16_t scancode            = 0;
+static uint8_t  alarm_cnt           = 0;
+
+HSV hsv;
+
+void led_test(uint8_t color);
+void clear_eeprom(void);
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+dprintf("keycode = 0x%X,TG(WIN_B) = 0x%X,TG(MAC_B) = 0x%X\r\n",keycode,TG(WIN_B),TG(MAC_B));
+//    uint8_t s = rgblight_get_sat();
+ //   uint8_t v = rgblight_get_val();
+
+    switch (keycode) {
+        case MO(WIN_FN):
+        case MO(MAC_FN):
+            fn_make_flag = false;
+            if (record->event.pressed) {
+                fn_make_flag = true;
+            } 
```

Comment:
```suggestion
            fn_make_flag = record->event.pressed;
```

Alternatively, you could use: 

```c
layer_state_t layer_state_set_kb(layer_state_t state) {
    fn_make_flag = (layer_state_cmp(state, WIN_FN) || layer_state_cmp(state, MAC_FN));
    return layer_state_set_user(state);
}
```

Reply:
ok,I took the first change,thx